src/main/java/com/minecolonies/blockout/views/DropDownList.java

```diff
@@ -0,0 +1,308 @@
+package com.minecolonies.blockout.views;
+
+import com.minecolonies.blockout.controls.Button;
+import com.minecolonies.blockout.controls.ButtonVanilla;
+import com.minecolonies.blockout.controls.Label;
+import com.minecolonies.blockout.views.ScrollingList;
+import com.minecolonies.blockout.OverlayView;
+import com.minecolonies.blockout.Pane;
+import com.minecolonies.blockout.PaneParams;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A DropDownList is a Button which when click display a ScrollingList below it.
+ */
+//TODO do not extend ButtonVanilla but extend view
+public class DropDownList extends ButtonVanilla implements Button.Handler
+{
+    /**
+     * View in which the list will be displayed.
+     */
+    protected OverlayView   overlay;
+
+    /**
+     * List to choose from.
+     */
+    protected ScrollingList list;
+    /**
+     * date required to fill the list.
+     */
+    protected DataProvider  dataProvider;
+
+    /**
+     * handler of the onSelectedItemChanged.
+     */
+    protected Handler       handler;
+
+    /**
+     * width of the scrolling list, by default it is the same as the DropDownList width.
+     */
+    protected int           dropDownWidth;
+
+    /**
+     * maximum height of the scrolling list, by default it is the same as the DropDownList width.
+     */
+    protected int           dropDownHeight;
+
+    /**
+     * index of the selected item.
+     */
+    protected int selectedIndex = -1;
+
+    /**
+     * Default constructor required by Blockout.
+     */
+    public DropDownList()
+    {
+        super();
+    }
+
+    /**
+     * Constructs a DropDownList from PaneParams.
+     *
+     * @param params Params for the ScrollingList
+     */
+    public DropDownList(final PaneParams params)
+    {
+        super(params);
+        final PaneParams.SizePair dropDownSize = params.getSizePairAttribute("dropDownSize", null, null);
+        dropDownWidth = dropDownSize == null ? width : dropDownSize.getX();
+        //When unknown, we use the same height as it is wide.
+        dropDownHeight = dropDownSize == null ? width : dropDownSize.getY();
+
+        overlay = new OverlayView();
+        overlay.setVisible(false);
+        overlay.setPosition(0, 0);
+
+        list = new ScrollingList(params);
+        list.setSize(dropDownWidth, dropDownHeight);
+        list.setPosition((x + width / 2) - dropDownWidth / 2, y + height);
+        list.putInside(overlay);
+        list.parseChildren(params);
+
+        setHandler(this);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Remove this call from a constructor to the overridable "setHandler" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)